src/components/AddToCart.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { selectProductInCart } from "../store/users/selectors";
+import { changeProductAmountInCart } from "../store/users/actions";
+import "./AddToCart.scss";
+
+export default function AddToCart({ productId }) {
+  const [amount, setAmount] = useState("");
+  const productInCart = useSelector(selectProductInCart(1, productId));
+  const dispatch = useDispatch();
+
+  return (
+    <div className="addToCart">
+      <div className="col1">
+        In cart: {productInCart ? productInCart.amount : 0}
+      </div>
+      <div className="col2">change to:</div>
+      <div className="col3">
+        <input
+          type="number"
+          min={0}
+          max={999}
+          value={amount}
+          onChange={(e) => {
+            setAmount(Number(e.target.value));
+            if (e.target.value > 999) {
+              setAmount(999);
+            }
```

Comment:
There's a chance here you are calling `setAmount` twice, that could generate some issues. I would switch it to an `if/else`